tlair/src/main/scala/at/forsyte/apalache/tla/lir/oper/TlaOper.scala

```diff
@@ -36,6 +38,37 @@ trait TlaOper extends Serializable {
 }
 
 object TlaOper {
+  /**
+    * Group a list of arguments into two lists: even and odd. The indices are starting with 0.
+    * @param args a list of expressions
+    * @return a pair of lists: on even positions and on odd positions
+    */
+  def deinterleave(args: Seq[TlaEx]): (Seq[TlaEx], Seq[TlaEx]) = {
```

Comment:
A more general question: Why don't we just represent these kinds of structures with a list of tuples instead of a list of altering elements? Seems like it would save a lot of conversions.

Reply:
I would say, it was a bit of overcompensation on our side. The syntax tree in SANY is quite baroque. So we wanted to have the simplest representation possible. On the other hand, there is a lot of code that processes arguments as lists, without knowing the argument structure of every particular operator. If we introduce tuples, that code will become more complex.

Reply:
Hmm, I'm skeptical whether this is the best representation :D But I'll withold judgment until I am more familiar with our context and code base :)